pymeasure/instruments/srs/ldc500series.py

```diff
@@ -0,0 +1,516 @@
+        super().__init__(adapter, name, **kwargs)
+
+    @property
+    def options(self):
+        """Get options not implemented, raises ``NotImplementedError``"""
+        raise NotImplementedError("options not implemented in LDC500series.")
+
+    @property
+    def next_error(self):
+        """Get next error not implemented, raises: ``NotImplementedError``"""
+        raise NotImplementedError("next_error not implemented in LDC500series.")
+
+    last_execution_error = Instrument.measurement(
+        "LEXE?",
+        """Get the last execution error code. This also resets the execution error code to 0.""",
+        cast=int,
+    )
+
+    last_command_error = Instrument.measurement(
+        "LCME?",
+        """Get the last command error code. This also resets the execution error code to 0.""",
+        cast=int,
+    )
+
+    def check_errors(self):
+        """Read all errors from the instrument.
+
+        :return: List of [``last_execution_error``, ``last_command_error``]
+        """
+        return [self.last_execution_error, self.last_command_error]
+
+    def check_set_errors(self):
+        """Check for errors after having set a property, and raise an error if any are present."""
+        # FIXME: The LDC500 has a delay between a command being executed and an error being logged.
+        #        *OPC? didn't work to wait for the previous command, so instead as short a delay
+        #        as possible has been put in. It's not a nice solution and any ideas would be
+        #        appreciated.
+        sleep(0.015)
+        errors = self.check_errors()
+        if errors == [0, 0]:
+            return []
+        raise Error(f"Error setting value: {errors}")
+
+
+class LDC500Series(LDC500SeriesBase):
+    """Represents an SRS LDC500Series laser diode controller
+    and provides a high-level interface for interacting with the instrument.
+
+    Attributes
+    ----------
+    ld : LDC500SeriesLDSubsystem
+        Provides access to laser diode control (e.g. ``ldc.ld.mode``)
+    pd : LDC500SeriesPDSubsystem
+        Provides access to photodiode control (e.g. ``ldc.pd.power``)
+    tec : LDC500SeriesTECSubsystem
+        Provides access to TEC control (e.g. ``ldc.tec.current``)
+
+    Example
+    ~~~~~~~
+
+    .. code-block:: python
+
+        ldc = LDC500Series("GPIB::5")
+        ldc.ld.enabled = True
+        temperature = ldc.tec.temperature
+
+    Glossary:
+    ~~~~~~~~~
+
+        - CC: Constant Current Mode
+        - CP: Constant Power Mode
+        - CT: Constant Temperature Mode
+        - LD: Laser diode
+        - PD: Photodiode
+        - TEC: Thermo-electric controller
+    """
+
+    def __init__(self, adapter, name="LDC500Series", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.ld = LDC500SeriesLDSubsystem(self)
+        self.pd = LDC500SeriesPDSubsystem(self)
+        self.tec = LDC500SeriesTECSubsystem(self)
+
+
+class LDC500SeriesLDSubsystem(LDC500SeriesBase):
```

Comment:
Should be based on `Channel` (or some class inheriting from `Channel`), as every `Instrument` should have its own adapter. All composite classes should be Channels.